jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/deploy/DeployServerlessApplicationDialog.kt

```diff
@@ -223,36 +225,22 @@ class DeployServerlessApplicationDialog(
             }
 
             // stack parameters
-            row {
-                cell(isFullWidth = true) {
-                    if (ApplicationManager.getApplication().isUnitTestMode) {
-                        // dialog can't be created in test mode since there's a lot of magic happening to the dialog, so insert a dummy to test validation logic
-                        label("dummy").withValidationOnApply { validateParameters() }
-                        return@cell
-                    }
+            row(message("serverless.application.deploy.template.parameters")) {
+                parametersField()
+                    .withBinding(::templateParameters.toBinding())
+                    .toolTipText(message("serverless.application.deploy.tooltip.template.parameters"))
+                    .withValidationOnApply { validateParameters() }
+            }
 
-                    val tableComponent = environmentVariablesTable.component
-                    ToolbarDecorator.findAddButton(tableComponent)?.isVisible = false
-                    ToolbarDecorator.findRemoveButton(tableComponent)?.isVisible = false
-                    ToolbarDecorator.findEditButton(tableComponent)?.isVisible = false
-
-                    tableComponent()
-                        .withBinding(
-                            componentGet = { _ ->
-                                environmentVariablesTable.stopEditing()
-                                environmentVariablesTable.environmentVariables
-                            },
-                            componentSet = { _, value ->
-                                environmentVariablesTable.setValues(value)
-                            },
-                            ::templateParameters.toBinding()
-                        )
-                        .toolTipText(message("serverless.application.deploy.tooltip.template.parameters"))
-                        .applyToComponent {
-                            border = IdeBorderFactory.createTitledBorder(message("serverless.application.deploy.template.parameters"), false)
-                        }
-                        .withValidationOnApply { validateParameters() }
-                }
+            // deploy tags
+            val tagsString = message("tags.title")
+            row(tagsString) {
+                tagsField()
+                    .withBinding(::tags.toBinding())
+                    .withValidationOnApply {
+                        // TODO: https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html#tag-conventions
+                        null
+                    }
```

Comment:
Not sure we should check this client side....should we just remove it if we never are going to do it?